src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -612,15 +612,40 @@ class ContentSidebar extends PureComponent<Props, State> {
         }
     }
 
+    /**
+     * Adds a comment to the comments state and increases total_count.
+     *
+     * @param {Comment} comment - The newly created comment from the API
+     * @return {void}
+     */
+    onCommentCreateSuccess(comment: Comment): void {
+        const { comments } = this.state;
+        if (comments && comments.entries) {
```

Comment:
Right now you will have to pass in comments to the API. 👎 

But in the future, your API code should be like

```js
// create comment
const comments = this.cache.get('comments_123');
comments.total_count += 1;
comments.entries.push(comment);
this.cache.set('comments_123', comments);

// fetch comments
...
if (this.cache.has('comment_123')) {
    return this.cache.get('comments_123');
}
...

// onCommentCreate
this.api.getCommentsAPI(false).createComment({
    successCallback: fetchComments
}
```

Reply:
@phora, this isn't passing comments into the API, though. This is just updating the `comments` state to include the newly created comment. Are you suggesting all feed item types that are currently in `state` (`versions`, `tasks`, `comments`) be stored in their respective API layers? If so, this will require a bigger refactor than we should be doing in this PR